--- libpng-1.2.9/scripts/makefile.gcmmx.orig	2006-04-17 01:12:53.813611500 +0200
+++ libpng-1.2.9/scripts/makefile.gcmmx	2006-04-17 01:13:47.872990000 +0200
@@ -72,8 +72,8 @@
 #	-m486 -malign-double -ffast-math \
 #	$(ALIGN) -funroll-loops -funroll-all-loops -fomit-frame-pointer 
 
-LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng12 -lz -lm
-LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a -lz -lm
+LDFLAGS=-L. -lpng12 -lz -lm
+LDFLAGS_A=libpng.a -lz -lm
 
 
 INCPATH=$(prefix)/include
@@ -122,7 +122,7 @@
 	echo I_opts=\"-I$(INCPATH)/$(LIBNAME)\"; \
 	echo cppflags=\"-DPNG_THREAD_UNSAFE_OK -DPNG_USE_PNGGCCRD\"; \
 	echo L_opts=\"-L$(LIBPATH)\"; \
-	echo R_opts=\"-Wl,-rpath,$(LIBPATH)\"; \
+	echo R_opts=\"\"; \
 	echo libs=\"-lpng12 -lz -lm\"; \
 	cat scripts/libpng-config-body.in ) > libpng-config
 	chmod +x libpng-config
@@ -239,7 +239,6 @@
 test-installed:
 	$(CC) -I$(ZLIBINC) \
 	   `$(BINPATH)/$(LIBNAME)-config --cflags` pngtest.c \
-	   -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) \
 	   -o pngtesti `$(BINPATH)/$(LIBNAME)-config --ldflags`
 	./pngtesti pngtest.png
 
--- libpng-1.2.9/scripts/makefile.linux.orig	2006-04-17 01:12:53.813611500 +0200
+++ libpng-1.2.9/scripts/makefile.linux	2006-04-17 01:14:15.322705500 +0200
@@ -50,8 +50,8 @@
 CFLAGS=-I$(ZLIBINC) -Wall $(OPT_FLAGS) -funroll-loops \
 	$(ALIGN) # $(WARNMORE) -g -DPNG_DEBUG=5
 
-LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng12 -lz -lm
-LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a -lz -lm
+LDFLAGS=-L. -lpng12 -lz -lm
+LDFLAGS_A=libpng.a -lz -lm
 
 INCPATH=$(prefix)/include
 LIBPATH=$(exec_prefix)/lib
@@ -98,7 +98,7 @@
 	echo prefix=\"$(prefix)\"; \
 	echo I_opts=\"-I$(INCPATH)/$(LIBNAME)\"; \
 	echo L_opts=\"-L$(LIBPATH)\"; \
-	echo R_opts=\"-Wl,-rpath,$(LIBPATH)\"; \
+	echo R_opts=\"\"; \
 	echo libs=\"-lpng12 -lz -lm\"; \
 	cat scripts/libpng-config-body.in ) > libpng-config
 	chmod +x libpng-config
@@ -207,7 +207,6 @@
 test-installed:
 	$(CC) -I$(ZLIBINC) \
 	   `$(BINPATH)/$(LIBNAME)-config --cflags` pngtest.c \
-	   -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) \
 	   -o pngtesti `$(BINPATH)/$(LIBNAME)-config --ldflags`
 	./pngtesti pngtest.png
 
